pkgs/development/tools/mold/default.nix

```diff
@@ -30,6 +31,35 @@ stdenv.mkDerivation rec {
   LTO = 1;
   makeFlags = [ "PREFIX=${placeholder "out"}" ];
 
+  suffixSalt = lib.replaceStrings ["-" "."] ["_" "_"] stdenv.targetPlatform.config;
+  wrapperName = "MOLD_WRAPPER";
+  coreutils_bin = lib.getBin coreutils;
+  postInstall = let
+    targetPrefix = lib.optionalString (stdenv.targetPlatform != stdenv.hostPlatform)
+                                          (stdenv.targetPlatform.config + "-");
+  in ''
+    mkdir -p $out/nix-support
+
+    mv $out/bin/mold $out/bin/.mold
+
+    export prog=$out/bin/.mold
```

Comment:
Why not use wrapProgram or makeWrapper?

Reply:
i don't understand how those tools would help solve the problem in a better way.

Reply:
I don't think this would help here.